swift/app.py

```diff
@@ -38,3 +40,43 @@ def frames(self) -> Iterable[QWidget]:
             An iterable object of Frame objects for showing.
         """
         return ()
+
+    def broadcast(self, channelName: str, content: Any):
+        """Broadcasts the content to the target channel.
+
+        Args:
+            channelName: Target channel name.
+            content: Content to be broadcast. It should be able to be converted to JSON object.
+        """
+        try:
+            msg = json.dumps(content)
+        except TypeError as e:
+            print(f"swift.app.broadcast(): {e!r}")
+        else:
+            self.broadcastRequested.emit(channelName, msg)
+
+    def receivedSlot(self, channelName: str, content: Any):
+        """Handles the received broadcast message.
+        
+        This is called when self.received signal is emitted.
+        This will be overridden by child classes.
+
+        Args:
+            channelName: Channel name that transferred the message.
+            content: Content to be broadcast.
```

Comment:
Since this is `received` slot, "received content" or something like that would be more appropriate.
"to be broadcast" looks like it transmits the content, rather that receives it.